test/components/views/rooms/DecryptionFailureBar-test.tsx

```diff
@@ -0,0 +1,371 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from "react";
+import { act, fireEvent, render, screen, waitFor, RenderResult } from "@testing-library/react";
+import "@testing-library/jest-dom";
+import { RoomKeyRequestState } from "matrix-js-sdk/src/crypto/OutgoingRoomKeyRequestManager";
+import { OutgoingRoomKeyRequest } from "matrix-js-sdk/src/crypto/store/base";
+
+import MatrixClientContext from "../../../../src/contexts/MatrixClientContext";
+import { DecryptionFailureBar } from "../../../../src/components/views/rooms/DecryptionFailureBar";
+
+type MockDevice = { device_id: string };
+
+const verifiedDevice1: MockDevice = { device_id: "verified1" };
+const verifiedDevice2: MockDevice = { device_id: "verified2" };
+const unverifiedDevice1: MockDevice = { device_id: "unverified1" };
+const unverifiedDevice2: MockDevice = { device_id: "unverified2" };
+
+const mockEvent1 = {
+    event: { event_id: "mockEvent1" },
+    getWireContent: () => ({ session_id: "sessionA" }),
+};
+
+const mockEvent2 = {
+    event: { event_id: "mockEvent2" },
+    getWireContent: () => ({ session_id: "sessionB" }),
+};
+
+const mockEvent3 = {
+    event: { event_id: "mockEvent3" },
+    getWireContent: () => ({ session_id: "sessionB" }),
+};
+
+const outgoingRequest3: OutgoingRoomKeyRequest = {
+    requestId: "outgoingRequest",
+    recipients: [],
+    requestBody: {
+        session_id: "sessionB",
+        sender_key: "senderkey",
+        room_id: "!room",
+        algorithm: "m.megolm.v1.aes-sha2",
+    },
+    state: RoomKeyRequestState.Sent,
+};
+
+const userId = "@user:example.com";
+
+let ourDevice: MockDevice | undefined;
+let allDevices: MockDevice[] | undefined;
+let keyBackup = false;
+
+const mockClient = {
+    getUserId: () => userId,
+    getDeviceId: () => ourDevice.device_id,
+    getDevices: () => Promise.resolve({ devices: allDevices }),
+    isSecretStored: jest.fn(() => Promise.resolve(keyBackup ? { key: "yes" } : null)),
+    checkIfOwnDeviceCrossSigned: (deviceId: string) => deviceId.startsWith("verified"),
+    downloadKeys: jest.fn(() => {}),
+    cancelAndResendEventRoomKeyRequest: jest.fn(() => {}),
+    getOutgoingRoomKeyRequest: jest.fn((ev) => Promise.resolve(ev === mockEvent3 ? outgoingRequest3 : null)),
+};
+
+function getBar(wrapper: RenderResult) {
+    return wrapper.container.querySelector(".mx_DecryptionFailureBar");
+}
+
+describe("<DecryptionFailureBar />", () => {
```

Comment:
I can't entirely figure out if this is a convention or not, but on balance it seems unusual to include the `< ... />`

```suggestion
describe("DecryptionFailureBar", () => {
```

Reply:
Our other unit tests for components also do this, see https://github.com/matrix-org/matrix-react-sdk/blob/develop/test/components/views/rooms/RoomPreviewBar-test.tsx#L68 for example.

Reply:
Yeah, there seems to be absolutely no convention here :man_shrugging: